### PR TITLE
test(api): add e2e tests for sql field auth apikey

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Add SQL field auth tests for API Key auth provider.

##### CDK / CloudFormation Parameters Changed
No

#### Issue #, if available
NA

#### Description of how you validated changes
- Manual testing
- E2E tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts [hunk 1]
```diff
@@ -173,4 +182,68 @@ describe('@auth directive without @model', () => {
     expectNoDirectives(queryType!);
     expectOneDirective(getField(queryType, 'getSecret'), 'aws_iam');
   });
+
```
Comment (palpatim) [inline packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts:185]: While you're in here, we should add a test case for SQL support for field-level auth on non-model types.

**Review by palpatim — APPROVED** [review]
Request for an additional test, but it can be added in a separate PR rather than blocking this one.

File: packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts (hunk 1 above)
Comment (sundersc) [reply packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts:185]: I will add in a follow-up PR.

**Comment by sundersc** [comment]
These changes are included in PR #2181.

**Pull request closed without merging**